### PR TITLE
refactor: email-theme public API + retire brand-template

### PR DESCRIPTION
**Pull request by webmarket7**
## Summary

Three-stage refactor that replaces the legacy `BrandStyleConfig` +
`brand-template.ts` branding pipeline with a theme-based RCML flow:

- **`@rule-io/rcml` email-theme public API** (\`1f3cc1b\`) — typed
  `EmailTheme` shape, `createEmailTheme`, `applyTheme` (overlay
  semantics), `getTheme` (inverse extractor). Public surface confined
  to the `email/` root; per-field upsert helpers live under
  `email/theme/` and are `@internal`.
- **`RuleClient` consumers on the theme path** (\`2d1f3ba\`) —
  `createAutomationEmail` / `createCampaignEmail` now build their
  auto-generated RCML via `emailThemeFromBrandStyle` + `applyTheme`.
- **Full retirement + vendor / CLI migration** (\`5867ecd\`) —
  `brand-template.ts` (1,304 lines) + its 1,089-line test are
  deleted. `BrandStyleConfig` is retired.
  `VendorConsumerConfig.brandStyle: BrandStyleConfig` flips to
  `.theme: EmailTheme`. `EmailTheme` types relocate from rcml to
  core so `VendorConsumerConfig` can reference them without a core
  → rcml dependency cycle. A new shared `@rule-io/rcml/brand-elements`
  module exposes the theme-aware element builders consumed by every
  vendor package + the CLI (`brandHeading`, `brandText`,
  `brandButton`, `contentSection`, `logoSection`, `footerSection`,
  `brandLoop`, `statusTrackerSection`, `addressBlock`,
  `buildThemedDocument`, `accentBackground`, etc.). The three
  vendor packages all drop their `@rule-io/client` dep.

Net diff across the branch: −1,079 lines; the theme path ends up
shorter than the legacy layer it replaces.

## Test plan

- [ ] \`npx nx run-many -t build\` — all 8 projects compile
- [ ] \`npx nx run-many -t vitest:test\` — all suites green (rcml 1,059 ·
      client 208 · vendor-shopify 83 · vendor-bookzen 34 ·
      vendor-samfora 33 · core, cli, sdk unchanged)
- [ ] \`npx nx run-many -t eslint:lint\` — 0 errors across all 8
      projects
- [ ] Manual smoke test: run the CLI deploy commands against a
      sandbox account — \`deploy-shopify\`, \`deploy-bookzen\`,
      \`deploy-samfora\`, \`validate-rcml\` — confirm each still
      resolves a brand style, builds RCML, and creates the resources
      end-to-end
- [ ] Manual smoke test: \`RuleClient.createAutomationEmail\` +
      \`createCampaignEmail\` with a real \`brandStyleId\` — confirm
      the editor-compatible template still renders with the expected
      brand assets (logo, fonts, colours) applied

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/rcml/src/email/brand-elements.ts [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+ * `customFields`. Throws `RuleConfigError` listing the missing entries.
+ */
+export function validateRequiredFields(
+  customFields: CustomFieldMap,
+  fieldNames: Record<string, string | undefined>,
+): void {
+  const missing: string[] = [];
+
+  for (const [logical, fieldName] of Object.entries(fieldNames)) {
+    if (fieldName !== undefined && customFields[fieldName] === undefined) {
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/rcml/src/email/brand-elements.ts]: Made strict in b07f965 — `validateRequiredFields` now reports `fieldName === undefined` as missing ("${logical} (no field mapping configured)") and the docstring documents the new contract. This matches the existing strict copy in `vendor-samfora/src/automations.ts`; the silent skip was contradicting the docstring and letting non-null assertions blow up downstream. Added unit tests covering all four cases.

File: packages/client/src/brand-style-to-theme.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+}
+
+function mapLinks(brandLinks: RuleBrandStyleLink[]): EmailThemeSocialLink[] {
+  const out: EmailThemeSocialLink[] = [];
+
+  for (const link of brandLinks) {
+    const themeType = LINK_TYPE_TO_THEME_LINK[link.type];
+
+    if (themeType !== undefined) {
+      out.push({ type: themeType, url: link.link });
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/client/src/brand-style-to-theme.ts]: Filtered in b07f965 using `sanitizeUrl` from `@rule-io/core` (already the project pattern — `brand-elements.ts` uses it). Swept the same fix to `mapImages` (logo) and `mapFonts` (font URL) since `applyTheme` rejects unsafe URLs in all three places (`apply-theme.test.ts:219`, `theme-rcml.ts:762`). Tests cover empty / whitespace / `javascript:` / non-URL / valid-with-trim, asserting that unsafe slots fall back to theme defaults rather than overriding them with bad values.

File: packages/cli/src/commands/validate-rcml.ts [hunk 3]
```diff
@@ -681,18 +712,24 @@ async function create(): Promise<void> {
...
-  console.log(`  Section BG: ${brandStyle.sectionBackgroundColor}`);
-  console.log(`  Text:       ${brandStyle.textColor}`);
-  console.log(`  Social:     ${brandStyle.socialLinks?.length ?? 0} link(s)${brandStyle.socialLinks?.length ? ' — ' + brandStyle.socialLinks.map(l => l.name).join(', ') : ''}`);
+  console.log(`  ID:         ${theme.brandStyleId}`);
+  console.log(`  Logo:       ${theme.images.logo?.url ?? '(none)'}`);
+  console.log(`  Button:     ${theme.colors[EmailThemeColorType.Primary]?.hex}`);
+  console.log(`  Body BG:    ${theme.colors[EmailThemeColorType.Background]?.hex}`);
+  console.log(`  Section BG: ${theme.colors[EmailThemeColorType.Body]?.hex}`);
+  console.log(`  Text:       ${theme.colors[EmailThemeColorType.Secondary]?.hex}`);
+  const themeLinkEntries = Object.values(theme.links).filter(
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/cli/src/commands/validate-rcml.ts]: Fixed in b07f965 — `Text:` now reads `theme.fontStyles[Paragraph].color` (which carries the `dark` brand colour or default `#0F0F1F`). The accent slot is still useful, so kept it on its own line as `Brand:` instead of dropping it.

File: packages/templates/src/render.ts [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+}
+
+/** Evaluate with errors re-wrapped to carry the node path. */
+function safeEvaluate(expr: string, scopes: readonly Scope[], path: string): unknown {
+  try {
+    return evaluateExpression(expr, scopes)
+  } catch (err) {
+    if (err instanceof TemplateRenderError) {
+      throw new TemplateRenderError(err.message, path, err.source)
+    }
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/templates/src/render.ts:378]: Fixed in 554c775. Added a public `originalMessage` readonly field on `TemplateRenderError` so re-wrappers can pass the un-decorated message; `safeEvaluate` now uses it. Test in `render.test.ts` asserts the composed message has exactly one `(at ...)` suffix when an inner expression error is rewrapped.

File: packages/templates/src/expression.ts [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+
+        throw new TemplateRenderError(
+          `Cannot read property '${node.property}' of ${String(obj)} in '${source}'`,
+          '',
+          source,
+        )
+      }
+
+      return (obj as Record<string, unknown>)[node.property]
+    }
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/templates/src/expression.ts:542]: Hardened in 554c775. Both the `member` (line 521) and `index` (line 540) cases now route through a shared `readSafeProperty` helper that throws `TemplateRenderError` for `__proto__`, `constructor`, and `prototype` — closing the `obj.constructor.constructor("...")` Function-constructor escape path for both `.foo` and `obj[k]` access. New tests cover member+index variants of each forbidden key, plus a computed-key index case to make sure the denylist applies to dynamically-built strings.

File: packages/templates/src/expression.ts [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+          `Cannot read index of ${String(obj)} in '${source}'`,
+          '',
+          source,
+        )
+      }
+
+      const key = evalNode(node.index, scopes, source)
+
+      return (obj as Record<string, unknown>)[String(key)]
+    }
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/templates/src/expression.ts:560]: Fixed in 554c775 alongside the member-access thread — both go through the same `readSafeProperty` helper, and the `index` case stringifies the key first so `obj[k]` where `k = '__proto__'` is also blocked. Tests cover both.

File: packages/vendor-shopify/src/generic-templates.ts [hunk 7]
```diff
@@ -90,43 +112,41 @@ export interface WelcomeEmailConfig {
...
-    });
-
-    const socialLinks = brandStyle.socialLinks ?? [];
-    const socialElements = socialLinks
+    const socialElements = Object.values(theme.links)
+      .filter((link): link is NonNullable<typeof link> => link !== undefined)
       .map((link) => {
         try {
-          return createSocialChildElement({ attrs: { name: link.name, href: link.href } });
+          return createSocialChildElement({
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/vendor-shopify/src/generic-templates.ts]: Fixed in 554c775. Added `getConfiguredSocialLinks` to `@rule-io/rcml` (next to `DEFAULT_LINKS_MAP`) which drops any slot whose URL still equals the default placeholder. Used here, plus the matching site in `ecommerce-templates.ts` (`hasSocial` had the same bug) and the CLI `validate-rcml` report. Edge case where a consumer sets a URL to *exactly* the default string is documented in the helper's JSDoc — unlikely in practice since real social URLs include a username/path. Tests cover all-defaults, partial config, exact-default-collision, and missing slots.

File: packages/cli/src/commands/validate-rcml.ts [hunk 8]
```diff
@@ -577,14 +608,18 @@ function buildSectionGroups(
...
+    (l): l is NonNullable<typeof l> => l !== undefined,
+  );
+  const rawSocialLinks: { name: string; href: string }[] =
+    themeSocialLinks.length > 0
+      ? themeSocialLinks.map((l) => ({ name: l.type, href: l.url }))
+      : [
+          { name: 'facebook', href: 'https://facebook.com' },
+          { name: 'instagram', href: 'https://instagram.com' },
+          { name: 'x', href: 'https://x.com' },
+          { name: 'web', href: 'https://example.com' },
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/cli/src/commands/validate-rcml.ts]: Fixed in 554c775 — changed `web` to `website` to match the project convention (`DEFAULT_LINKS_MAP.website`, `LINK_TYPE_TO_THEME_LINK.website`).

File: packages/templates/package.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    ".": {
+      "types": "./src/index.d.ts",
+      "import": "./src/index.js"
+    }
+  },
+  "files": [
+    "src",
+    "README.md",
+    "CHANGELOG.md"
+  ],
```
[the remark this answers is not in the document]
Comment (swesam) [reply packages/templates/package.json:16]: Fixed in 554c775 — dropped `README.md` and `CHANGELOG.md` from `files` since neither exists in the package. `@rule-io/templates` will ship without docs on npm for now.
